ddtrace/contrib/unittest/patch.py

```diff
@@ -431,13 +536,64 @@ def handle_test_wrapper(func, instance, args: tuple, kwargs: dict):
         if test_suite_span is None and test_module_suite_path in _CIVisibility._unittest_data["suites"]:
             test_suite_span = _start_test_suite_span(instance)
-            _CIVisibility._unittest_data["suites"][test_module_suite_path]["suite_span"] = test_suite_span
+            suite_dict = _CIVisibility._unittest_data["suites"][test_module_suite_path]
+            suite_dict["suite_span"] = test_suite_span
         if not test_module_span or not test_suite_span:
             log.debug("Suite and/or module span not found for test: %s", test_name)
             return func(*args, **kwargs)
         with _start_test_span(instance, test_suite_span) as span:
-            result = func(*args, **kwargs)
+            test_session_span = _CIVisibility._datadog_session_span
+            root_directory = os.getcwd()
+            if _CIVisibility.test_skipping_enabled():
+                if _is_marked_as_unskippable(instance):
+                    span.set_tag_str(test.ITR_UNSKIPPABLE, "true")
+                    test_module_span.set_tag_str(test.ITR_UNSKIPPABLE, "true")
+                    test_session_span.set_tag_str(test.ITR_UNSKIPPABLE, "true")
+                test_name = _extract_test_method_name(instance)
+                test_module_suite_path_without_extension = "{}/{}".format(
+                    os.path.splitext(test_module_path)[0], test_suite_name
+                )
+                if (
+                    len(args)
+                    and _CIVisibility._instance._should_skip_path(test_module_suite_path_without_extension, test_name)
+                    and not _is_skipped_test(instance)
+                ):
+                    global _global_skipped_elements
+                    _global_skipped_elements += 1
+
+                    test_module_span._metrics[test.ITR_TEST_SKIPPING_COUNT] += 1
+                    test_module_span.set_tag_str(test.ITR_TEST_SKIPPING_TESTS_SKIPPED, "true")
+                    test_module_span.set_tag_str(test.ITR_DD_CI_ITR_TESTS_SKIPPED, "true")
+
+                    test_session_span.set_tag_str(test.ITR_TEST_SKIPPING_TESTS_SKIPPED, "true")
+                    test_session_span.set_tag_str(test.ITR_DD_CI_ITR_TESTS_SKIPPED, "true")
+
+                    if _is_marked_as_unskippable(instance):
+                        span.set_tag_str(test.ITR_FORCED_RUN, "true")
+                        test_module_span.set_tag_str(test.ITR_FORCED_RUN, "true")
+                        test_session_span.set_tag_str(test.ITR_FORCED_RUN, "true")
+                    else:
+                        instance._dd_itr_skip = True
+                        span.set_tag_str(test.ITR_SKIPPED, "true")
+                        span.set_tag_str(test.SKIP_REASON, SKIPPED_BY_ITR_REASON)
+
+            if _is_test_coverage_enabled(instance):
+                coverage = _start_coverage(root_directory)
+                instance._coverage = coverage
```

Comment:
We shouldn't run coverage for tests that ITR skips, so I'd move this to the `else` below.